### PR TITLE
Added `databricks_recipient` resource for Delta Sharing

### PR DESCRIPTION
**Pull request by stikkireddy**
- [x] docs
- [x] resource
- [x] tests
- [x] supports ip access lists
- [x] supports db 2 db
- [x] supports db 2 open

Examples:

```hcl
resource "databricks_recipient" "db2open" {
  name = "db2open-recipient"
  comment = "made by terraform"
  authentication_type = "TOKEN"
  sharing_code = "my code "
  ip_access_list {
    allowed_ip_addresses = [...] // fill out relevant IPv4 addresses (CIDR notation is accepted)
  }
}

resource "databricks_recipient" "db2db" {
  name = "sri-terraform-test-db2db-recipient"
  comment = "made by terraform"
  authentication_type = "DATABRICKS"
  data_recipient_global_metastore_id = "<cloud>:<region>:<guid>"
}
```

### CODE REVIEW

**Comment by alexott** [comment]
do we always need to include

```hcl
ip_access_list {
    allowed_ip_addresses = ["0.0.0.0/0"]
  }
```
?

File: docs/resources/recipient.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+# databricks_recipient Resource
+
+Within a metastore, Unity Catalog provides the ability to create a recipient to attach delta shares to.
+
+A `databricks_recipient` is contained within [databricks_metastore](metastore.md) and can contain a list of shares.
+
+## Example Usage
+
+```hcl
+resource "databricks_recipient" "db2open" {
```
Comment (nfx) [inline docs/resources/recipient.md:13]: @wchau - what do you think about `databricks_catalog_recipient` or `databricks_share_recipient` as alternative logical names for this?

File: docs/resources/recipient.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  ip_access_list {
+    allowed_ip_addresses = ["0.0.0.0/0"]
+  }
+}
+
+resource "databricks_recipient" "db2db" {
+  name = "sri-terraform-test-db2db-recipient"
+  comment = "made by terraform"
+  authentication_type = "DATABRICKS"
+  data_recipient_global_metastore_id = "<cloud>:<region>:<guid>"
```
Comment (nfx) [inline docs/resources/recipient.md:27]: can you expand the example with `databricks_metastore` resource to refer the ID from it?

File: docs/resources/recipient.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+## Attribute Reference:
+
+* `tokens` - (Optional) List of Recipient Tokens.
+
+## Related Resources
+
+The following resources are used in the same context:
+
+* [databricks_table](../data-sources/tables.md) data to list tables within Unity Catalog.
```
Comment (nfx) [inline docs/resources/recipient.md:64]: can you add those in the later PRs? :)

**Review by nfx — CHANGES_REQUESTED** [review]
Suggested a resource rename and left comments about example expansion. Can you also add an acceptance test? e.g. assume that metastore will be provided as `TEST_DATABRICKS_METASTORE_ID` environment variable in the testing environment.

File: docs/resources/recipient.md (hunk 1 above)
Comment (wchau) [reply docs/resources/recipient.md:13]: `databricks_catalog_recipient` would not make sense in the Delta Sharing context. What's wrong with `databricks_recipient`? In UC and DBR, `recipient` is a top-level keyword.

**Comment by stikkireddy** [comment]
> do we always need to include
> 
> ```terraform
> ip_access_list {
>     allowed_ip_addresses = ["0.0.0.0/0"]
>   }
> ```
> 
> ?

AFAIK you don't @alexott it was just a poor example on my part. Ill change it to some other random public ip

File: catalog/resource_recipient.go [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+	Tokens                         []Token       `json:"tokens,omitempty" tf:"computed"`
+	DataRecipientGlobalMetastoreId string        `json:"data_recipient_global_metastore_id,omitempty" tf:"force_new,conflicts:ip_access_list"`
+	IpAccessList                   *IpAccessList `json:"ip_access_list,omitempty"`
+}
+
+type Recipients struct {
+	Recipients []RecipientInfo `json:"recipients"`
+}
+
+//func (a RecipientsAPI) list() (recipients Recipients, err error) {
```
Comment (nfx) [inline catalog/resource_recipient.go:47]: please remove commented code. we have a generated SDK for that already.

**Comment by nfx** [comment]
@stikkireddy i think it's better to remove `allowed_ip_addresses` line from examples and leave it as `// .. fill in allowed addresses`, because customers will expose the share to entire internet and would then complain to support about bad documentation.

File: docs/resources/recipient.md [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+Setting `authentication_type` type to `TOKEN` creates a temporary url to download a credentials file. This is used to 
+authenticate to the sharing server to access data. This is for when the recipient is not using Databricks. 
+
+```hcl
+resource "random_password" "db2opensharecode" {
+  length           = 16
+  special          = true
+}
+
+data databricks_current_user "current" {}
```
Comment (nfx) [inline docs/resources/recipient.md:23]:
```suggestion
data "databricks_current_user" "current" {}
```

File: docs/resources/recipient.md [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+```
+
+### Databricks to Databricks Sharing
+
+Setting `authentication_type` type to `DATABRICKS` allows you to automatically create a provider for a recipient who 
+is using Databricks. To do this they would need to provide the global metastore id that you will be sharing with. The 
+global metastore usually the format: `<cloud>:<region>:<guid>` 
+
+```hcl
+data databricks_current_user "current" {}
```
Comment (nfx) [inline docs/resources/recipient.md:43]:
```suggestion
data "databricks_current_user" "current" {}
```

File: docs/resources/recipient.md [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+    allowed_ip_addresses = [...] // .. fill in allowed IPv4 addresses (CIDR notation allowed)
+  }
+}
+```
+
+### Databricks to Databricks Sharing
+
+Setting `authentication_type` type to `DATABRICKS` allows you to automatically create a provider for a recipient who 
+is using Databricks. To do this they would need to provide the global metastore id that you will be sharing with. The 
+global metastore usually the format: `<cloud>:<region>:<guid>`
```
Comment (nfx) [inline docs/resources/recipient.md:40]: > usually the format

this reads strange

**Review by nfx — APPROVED** [review]

**Pull request merged**